cpp/src/strings/utilities.cuh

```diff
@@ -77,20 +78,28 @@ auto make_strings_children(SizeAndExecuteFunction size_and_exec_fn,
                            rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
                            cudaStream_t stream                 = 0)
 {
-  auto transformer =
-    thrust::make_transform_iterator(thrust::make_counting_iterator<size_type>(0), size_and_exec_fn);
   auto offsets_column =
-    make_offsets_child_column(transformer, transformer + strings_count, mr, stream);
-  auto d_offsets    = offsets_column->view().template data<int32_t>();
-  auto chars_column = create_chars_child_column(
-    strings_count, null_count, thrust::device_pointer_cast(d_offsets)[strings_count], mr, stream);
-  size_and_exec_fn.d_offsets = d_offsets;  // set the offsets
-  size_and_exec_fn.d_chars =
-    chars_column->mutable_view().template data<char>();  // fill in the chars
-  thrust::for_each_n(rmm::exec_policy(stream)->on(stream),
-                     thrust::make_counting_iterator<size_type>(0),
-                     strings_count,
-                     size_and_exec_fn);
+    make_numeric_column(data_type{INT32}, strings_count + 1, mask_state::UNALLOCATED, stream, mr);
+  auto offsets_view          = offsets_column->mutable_view();
+  auto d_offsets             = offsets_view.template data<int32_t>();
+  size_and_exec_fn.d_offsets = d_offsets;
+
+  std::unique_ptr<column> chars_column = nullptr;
+  while (true) {
```

Comment:
Rather than a while loop that runs 1.5 times, why not just wrap the call to `thrust::for_each_n` in a lambda and call it twice explicitly?   I think it might be clearer?

Reply:
This would be the same as calling the functor twice as in the original code. The lambda is inlined for each call as it appears in the source code. So calling it twice will inline the regex twice again.

Reply:
This does work! I misunderstood the comment.
I created a lambda to call `thrust::for_each_n` and called it twice thus removing the while-loop.
Definitely much cleaner. Hopefully more clear as well.